R/formatters_var_labels.R

```diff
@@ -0,0 +1,175 @@
+#' Return an object with a label attribute
+#'
+#' @param x an object
+#' @param label label attribute to to attached to  \code{x}
+#'
+#' @source This function was taken 1-1 from
+#' \href{https://cran.r-project.org/web/packages/formatters/index.html}{formatters} package, to reduce the complexity of
+#' the dependency tree.
+#'
+#' @seealso [formatters_var_labels()] [formatters_var_relabel()] [`formatters_var_labels<-`]
+#'
+#' @export
+#' @return \code{x} labeled by \code{label}. Note: the exact mechanism of labeling should be
+#' considered an internal implementation detail, but the label will always be retrieved via \code{obj_label}.
+#' @examples
+#' x <- formatters_with_label(c(1, 2, 3), label = "Test")
+#' obj_label(x)
+formatters_with_label <- function(x, label) {
+  obj_label(x) <- label
```

Comment:
@gogonzo do you think we also need to include `obj_label` from formatters 
https://github.com/insightsengineering/formatters/blob/ffa09bb46eb28a9e269847b918c14c719c5d28b5/R/generics.R#L261C1-L283C2 ? Looks like it's a wrapper around S3 and S4 at the same time, where for S4 it uses standardGeneric and for S3 it just calls `attr(x, "label")`. I think in teal packages we focus on S3, so rewriting `obj_label(x) <- label` to `attr(x, "label") <- label` should be enough